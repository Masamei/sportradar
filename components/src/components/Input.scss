.input-podium {
    height: 30px;
    width: 200px;
    font-size: 15px;

    background-color: #1F58CF;
    border-radius: 3px;
    padding: 1px;
    display: inline-block;


    box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.4);

    input {
        border-radius: 1px;
        background-color: #E3EAFF;
        border-width: 0px;
        padding: 5px;
        width: 100%;
        height: 100%;
        box-sizing: border-box;
    }

    input:focus {
        outline: none;
    }
}

.error {
    background-color: #D20300;

    input {
        background-color: #FFE5E5;
    }

    input:focus {
        outline: none;
    }
}
    