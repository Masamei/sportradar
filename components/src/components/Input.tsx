import React from 'react'
import './input.scss'

export default function InputPodium(
  {value, type, hasError, placeholder, onChange, onKeyDown}: 
  {value: string, type: string, hasError: boolean, placeholder?: string, onChange: (e: React.ChangeEvent<HTMLInputElement>) => void, onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement>) => void }
  ) {
  return (
    <div className={`input-podium ${hasError ? ' error' : ''}`} >
      <input type={type} value={value} onChange={onChange} placeholder={placeholder} onKeyDown={onKeyDown} />
    </div>
  )
}
