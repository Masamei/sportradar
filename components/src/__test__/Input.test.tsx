import React from 'react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'
import Input from "../components/Input"

test('demo', () => {
    expect(true).toBe(true)
})

test("Renders the input and checks the snapshot", () => {
    const handleChange = jest.fn()
    const comp = render(<Input value='' type='text' onChange={handleChange} hasError={false} />)
    expect(comp).toBeTruthy()
    expect(comp).toMatchSnapshot('inputDefaultSnapshot')    
})

test("Renders the input with text and checks the snapshot", () => {
    const handleChange = jest.fn()
    const comp = render(<Input value='This is me.' onChange={handleChange} type='text' hasError={false} />)
    expect(comp).toBeTruthy()
    expect(comp).toMatchSnapshot('inputValueSnapshot')   
})

test("Renders the input with numbers and error and checks the snapshot", () => {
    const handleChange = jest.fn()
    const comp = render(<Input value='123' onChange={handleChange} type='text' hasError={true} />)
    expect(comp).toBeTruthy()
    
    const errorClassName = comp.container.querySelector('.error')
    expect(errorClassName).toBeTruthy()

    expect(comp).toMatchSnapshot('inputErrorSnapshot')   
})