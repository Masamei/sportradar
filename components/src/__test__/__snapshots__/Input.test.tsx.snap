// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the input and checks the snapshot: inputDefaultSnapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input-podium "
      >
        <input
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-podium "
    >
      <input
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders the input with numbers and error and checks the snapshot: inputErrorSnapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input-podium  error"
      >
        <input
          type="text"
          value="123"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-podium  error"
    >
      <input
        type="text"
        value="123"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders the input with text and checks the snapshot: inputValueSnapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input-podium "
      >
        <input
          type="text"
          value="This is me."
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-podium "
    >
      <input
        type="text"
        value="This is me."
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
